### ISSUE TITLE
Why updating the gradients of the guiding network C while training G?

### ISSUE THREAD

**Issue by ammar-deep** [clovaai/tunit#27]
Hi. I couldn't find the explanation of the following question on your paper. 

While training the Generator, could you please explain why you update the gradients of the guiding network C? (https://github.com/clovaai/tunit/blob/49e633179c5670228a2319d2c99054322616b516/train/train_unsupervised.py#L174-L180) Even though we are not backpropagating for the guiding network?

**Comment by FriedRonaldo** [comment]
The backward operation of g_loss calculates the gradients of g_loss of the guiding network because it works as a style encoder of the generator in lines 148 to 149. 

If the gradients do not update the guiding network, it is configuration F in the main text and the difference between F and G is described in the appendix (appendix 4).

Please refer to [Appendix](https://openaccess.thecvf.com/content/ICCV2021/supplemental/Baek_Rethinking_the_Truly_ICCV_2021_supplemental.pdf).

**Comment by ammar-deep** [comment]
Oh I missed the Appendix. Thanks for the quick response.
